src/imitation/data/types.py

```diff
@@ -75,7 +240,8 @@ def __eq__(self, other) -> bool:
         if not isinstance(other, Trajectory):
             return False
 
-        dict_self, dict_other = dataclasses.asdict(self), dataclasses.asdict(other)
+        dict_self = dataclass_quick_asdict(self)
```

Comment:
`dataclasses.asdict` is recursive which unwraps dictobs, so changing to `dataclass_quick_asdict(self)` in several places.